frontend/src/host_orchestrator/orchestrator/controller.go

```diff
@@ -37,6 +37,7 @@ import (
 
 const HeaderBuildAPICreds = "X-Cutf-Host-Orchestrator-BuildAPI-Creds"
 const HeaderUserProject = "X-Cutf-Host-Orchestrator-BuildAPI-Creds-User-Project-ID"
+const HeaderUseHostBuildAPICreds = "X-Cutf-Host-Orchestrator-Use-Host-BuildAPI-Creds"
```

Comment:
why is this header needed? why is it not the default when no other credential headers are present?

Reply:
Whether uncommon or not, if the host has credentials authorized with the build api it should use them by default, the client doesn't need to know this.

The underlying problem here is that all GCE hosts have credentials, `cvd` always finds those credentials and doesn't check whether they have access to the build api or not and the build api denies access to public builds for credentials that don't have access.

The obvious solution would be to fix the build api, but that's beyond our control. I agree that the host orchestrator must be told to use the GCE credentials, but it should not be the client who decides that. There should be some configuration file in the GCE image or a metadata key that HO can read that tells it to use the GCE credentials or not.